### ISSUE TITLE
WeChat app not opening

### ISSUE THREAD

**Issue by nlasagni** [Adyen/adyen-android#574]
**Describe the bug**
I integrated the WeChat Pay component and implemented my custom `WeChatPayActivity` that is created when the user selects the WeChat Pay payment method and should redirect to the WeChat app to perform payments.

The Activity is defined in the Manifest as follows:

```
<activity
    android:name=".ui.wechatpay.WeChatPayActivity"
    android:exported="true"
    android:launchMode="singleTop"
    android:screenOrientation="portrait"/>
```

After calling `weChatPayActionComponent.handleAction(this, action)` method nothing happens, the user is not switched to the WeChat app. 

Instead, the `onNewIntent` of my Activity is called immediately, and after calling the `weChatPayActionComponent.handleIntent(intent)` method, the `weChatPayActionComponent observer` is triggered with an `ActionComponentData` like the following: `{ "resultCode": -1 }`.

**To Reproduce**
Follow the Android WeChat component integration steps at https://docs.adyen.com/payment-methods/wechat-pay/wechat-pay-in-app-payments/android-component#handle-the-app-switch.

**Expected behavior**
The `WeChatPayActionComponent` should switch the user to the WeChat app when calling `weChatPayActionComponent.handleAction(this, action)`.

**Logs**
```
2022-01-12 16:23:27.642 19750-19750/.sample D/MicroMsg.SDK.WXMsgImplComm: pass
2022-01-12 16:23:27.642 19750-19750/.sample I/MicroMsg.SDK.WXApiImplV10: sendReq, req type = 5
2022-01-12 16:23:27.643 19750-19750/.sample D/MicroMsg.SDK.MMessageAct: send, targetPkgName = com.tencent.mm, targetClassName = com.tencent.mm.plugin.base.stub.WXPayEntryActivity
2022-01-12 16:23:27.657 19750-19750/.sample D/MicroMsg.SDK.MMessageAct: send mm message, intent=Intent { flg=0x18000000 cmp=com.tencent.mm/.plugin.base.stub.WXPayEntryActivity (has extras) }
2022-01-12 16:23:27.872 19750-19750/.sample I/slpayment: WeChatPayActivity onNewIntent Intent { flg=0x10400000 cmp=.sample/.ui.wechatpay.WeChatPayActivity (has extras) }
2022-01-12 16:23:27.874 19750-19750/.sample I/MicroMsg.SDK.WXApiImplV10: handleIntent, cmd = 5
```

**Smartphone (please complete the following information):**
 - Device: Xiaomi Mi A1
 - OS: Android 9

**Adyen version used**
- 4.2.0

**Comment by caiofaustino** [comment]
Hey there,

` { "resultCode": -1 }` usually means there was an error on the WeChat side and it's returning an error. 
Is the WeChat app installed? Is the account logged in with payment capabilities? Unfortunately we don't have much insight on the error that happened, as you can see `-1` doesn't tell much.

**Comment by nlasagni** [comment]
Hi @caiofaustino, thank you for your reply.

The WeChat app is installed, but maybe the thing is that my account doesn't have payment capabilities. Unfortunately, I saw from different web resources that currently is not easy to enable WeChat Pay and test it with Adyen from mobile.

Another thing is that when I log out from the WeChat app, I am able to open the WeChat app through Adyen SDK, and the app prompts the login screen.

Anyway, in December 2021 the I was able to redirect to the WeChat app while being logged in with the same WeChat account and code implementation that I have now.

As you said, is for sure an issue on the WeChat side, thank you for the support anyway.
